irrep/spacegroup.py

```diff
@@ -841,7 +848,7 @@ def __init__(
                 translations = dataset['translations']
             else:
                 self.name = dataset.international
-                self.number = dataset.number
+                self.number = str(dataset.number)
```

Comment:
Why does it have to be a 'str'?

Reply:
Magnetic SGs carry a dot in their number (e.g. 194.270) so they require a string representation. I standardized both MSGs and SGs to have the same `number` type.

Reply:
fixed. Now `SpaceGroup` has an attribute (property getter) `number` which returns the number of the space groups as an `int`, and `-1` if it is a magnetic space group. Internally, the attribute `number_str` is used.